### ISSUE TITLE
Mapfish print v2 in 1.6.0rc1

### ISSUE THREAD

**Issue by kalbermattenm** [camptocamp/cgxp#961]
I have several issues in 1.6.0rc1 without changing anything in the print...

First, in our Internet instance, the whole print process is working, but once the PDF retrieved, the print body mask stays stucked on the map:
![presse-papiers-1](https://cloud.githubusercontent.com/assets/1681332/8079210/6f3ab13c-0f63-11e5-8dc0-bf2f411d3ee5.jpg)

It also looks like we lost the default print translations...

Next, in our Intranet instance, the print Window doesn't even appear, because of an Ext error:
![presse-papiers-2](https://cloud.githubusercontent.com/assets/1681332/8079235/886e2be8-0f63-11e5-921e-343d2d58994e.jpg)

I don't know if that has something to do with the fact that in the Intranet instance we have an additional combo field ("Inclure un écusson communal") (see https://github.com/camptocamp/sitn_c2cgeoportal/blob/master/sitn/templates/viewer.js#L876-L940): 
![presse-papiers-3](https://cloud.githubusercontent.com/assets/1681332/8079270/a8e7d928-0f63-11e5-9405-8483a8ca0c3a.jpg)

**Comment by sbrunner** [comment]
There are some issue but the new configuration should be like this:
```
            additionalAttributes: [{
                name: "title",
                label: "Title",
                type: "String"
            }, {
                name: "comments",
                label: "Comments",
                type: "String",
                useTextArea: true
            }, {
                name: "legend",
                type: "LegendAttributeValue"
            }, {
                name: 'img_commune',
                type: 'Commune'
            }],
            addCustomItem: function(item, attribute, extraAttributes, fieldAttributes) {
                if (attribute == 'Commune') {
                    this.add({
                        xtype: 'combo',
                        fieldLabel: 'Inclure un écusson communal',
                        emptyText: 'Choisir',
                        name: 'img_commune',
                        typeAhead: true,
                        triggerAction: 'all',
                        lazyRender:true,
                        mode: 'local',
                        store: new Ext.data.ArrayStore({
                            id: 0,
                            fields: [
                                'myId',
                                'displayText'
                            ],
                            data: [
                                ["bevaix","Bevaix"],
                                ["boudry","Boudry"],
                                ["brot_dessous","Brot-Dessous"],
                                ["brot_plamboz","Brot-Plamboz"],
                                ["corcelles_cormondreche","Corcelles-Cormondrèche"],
                                ["cornaux","Cornaux"],
                                ["cortaillod","Cortaillod"],
                                ["cressier","Cressier"],
                                ["enges","Enges"],
                                ["fresens","Fresens"],
                                ["gorgier","Gorgier"],
                                ["hauterive","Hauterive"],
                                ["brevine","La Brévine"],
                                ["chaux_de_fonds","La Chaux-De-Fonds"],
                                ["chaux_du_milieu","La Chaux-Du-Milieu"],
                                ["cote_aux_fees","La Cote-Aux-Fées"],
                                ["sagne","La Sagne"],
                                ["tene","La Tène"],
                                ["cerneux_pequignot","Le Cerneux-Péquignot"],
                                ["landeron","Le Landeron"],
                                ["locle","Le Locle"],
                                ["brenets","Les Brenets"],
                                ["planchettes","Les Planchettes"],
                                ["ponts_de_martel","Les Ponts-De-Martel"],
                                ["verrieres","Les Verrières"],
                                ["lignieres","Lignières"],
                                ["milvignes","Milvignes"],
                                ["montalchez","Montalchez"],
                                ["neuchatel","Neuchâtel"],
                                ["peseux","Peseux"],
                                ["rochefort","Rochefort"],
                                ["saint_aubin_sauges","Saint-Aubin-Sauges"],
                                ["saint_blaise","Saint-Blaise"],
                                ["val_de_ruz","Val-De-Ruz"],
                                ["val_de_travers","Val-De-Travers"],
                                ["valangin","Valangin"],
                                ["vaumarcus","Vaumarcus"]
                            ]
                        }),
                        valueField: 'myId',
                        displayField: 'displayText'
                    }
                }
             }
```

**Comment by kalbermattenm** [comment]
Thanks @sbrunner, I will try this out tomorrow !

**Comment by kalbermattenm** [comment]
I get a Tomcat error (with or without the additional fields):

```
org.mapfish.print.JsonMissingException: attribute [spec.pages[0].comment] missing
        at org.mapfish.print.utils.PJsonObject.getString(PJsonObject.java:56)
        at org.mapfish.print.PDFUtils.getContextValue(PDFUtils.java:524)
        at org.mapfish.print.PDFUtils.renderString(PDFUtils.java:430)
        at org.mapfish.print.config.layout.TextBlock.render(TextBlock.java:46)
        at org.mapfish.print.PDFUtils.createCell(PDFUtils.java:624)
        at org.mapfish.print.PDFUtils.buildTable(PDFUtils.java:609)
        at org.mapfish.print.config.layout.ColumnsBlock$1.render(ColumnsBlock.java:58)
        at org.mapfish.print.PDFCustomBlocks.addAbsoluteDrawer(PDFCustomBlocks.java:183)
        at org.mapfish.print.config.layout.ColumnsBlock.render(ColumnsBlock.java:56)
        at org.mapfish.print.config.layout.Page.render(Page.java:73)
        at org.mapfish.print.config.layout.MainPage.render(MainPage.java:63)
        at org.mapfish.print.config.layout.Layout.render(Layout.java:59)
        at org.mapfish.print.output.AbstractOutputFormat.doPrint(AbstractOutputFormat.java:30)
        at org.mapfish.print.output.PdfOutputFactory.print(PdfOutputFactory.java:58)
        at org.mapfish.print.MapPrinter.print(MapPrinter.java:173)
        at org.mapfish.print.servlet.MapPrinterServlet.doCreatePDFFile(MapPrinterServlet.java:367)
        at org.mapfish.print.servlet.MapPrinterServlet.createPDF(MapPrinterServlet.java:184)
        at org.mapfish.print.servlet.MapPrinterServlet.doPost(MapPrinterServlet.java:112)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:641)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:722)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:304)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:210)
        at com.codahale.metrics.servlet.AbstractInstrumentedFilter.doFilter(AbstractInstrumentedFilter.java:97)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:243)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:210)
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:224)
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:185)
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:472)
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:151)
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:100)
        at org.apache.catalina.valves.AccessLogValve.invoke(AccessLogValve.java:929)
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:118)
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:405)
        at org.apache.coyote.http11.Http11Processor.process(Http11Processor.java:269)
        at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:515)
        at org.apache.tomcat.util.net.JIoEndpoint$SocketProcessor.run(JIoEndpoint.java:300)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
```
Moreover, I tried to include the additional field value, but I cannot see anything about it in the json parameters (the additional field is showing up in the panel, but the plugin's default translations are still missing , see second image hereunder):

![presse-papiers-1](https://cloud.githubusercontent.com/assets/1681332/8249741/92ac6096-166d-11e5-8ce2-6ce8ee4c4ae4.png)

![presse-papiers-2](https://cloud.githubusercontent.com/assets/1681332/8249758/ca756db0-166d-11e5-8faf-57426a182f95.png)

**Comment by sbrunner** [comment]
In the json you have `comments` and in the template `comment` without any s ...

**Comment by kalbermattenm** [comment]
Oups... so that changed between 1.5.x and 1.6rc1 (maybe it should be added in the `CONST_CHANGELOG.txt` file)

For the aditional field, it is still not showing up in the json sent to the print proxy. I made a PR so that you can see how I tried to add it in the `viewer.js`: https://github.com/camptocamp/sitn_c2cgeoportal/pull/952

Am I doing something wrong ?

**Comment by sbrunner** [comment]
No it's `comment` on boot it's the config I post who introduce an `s`...

**Comment by kalbermattenm** [comment]
OK, thanks ! I see your point.

Still the issue with the additional field...

**Comment by sbrunner** [comment]
Witch issue... I lose the thread...

**Comment by kalbermattenm** [comment]
Yeah... I can understand that...

I use the `addCustomItem` method in the `cgxp_print` plugin in my `viewer.js` file.

The additional field created through that appears in the print panel when clicking on the print button, but nothing is inserted in the post request made to the c2cgeoportal print proxy. 

* In c2cgeoportal 1.5.x, when a value is selected in this additional combobox, a `img_commune` key is added in the sent json.
* In c2cgeoportal 1.6, nothing is added to the sent json. It is as this field is not taken into account when the print plugin creates the posted json.

**Comment by sbrunner** [comment]
In my example I miss to to take the item properties ... =>
```
            additionalAttributes: [{
                name: "title",
                label: "Title",
                type: "String"
            }, {
                name: "comment",
                label: "Comment",
                type: "String",
                useTextArea: true
            }, {
                name: "legend",
                type: "LegendAttributeValue"
            }, {
                name: 'img_commune',
                type: 'Commune'
            }],
            addCustomItem: function(item, attribute, extraAttributes, fieldAttributes) {
                if (attribute.type == 'Commune') {
                    this.add(Ext.apply(item, {
                        xtype: 'combo',
                        fieldLabel: 'Inclure un écusson communal',
                        emptyText: 'Choisir',
                        typeAhead: true,
                        triggerAction: 'all',
                        lazyRender:true,
                        mode: 'local',
                        store: new Ext.data.ArrayStore({
                            id: 0,
                            fields: [
                                'myId',
                                'displayText'
                            ],
                            data: [
                                ["bevaix","Bevaix"],
                                ["boudry","Boudry"],
                                ["brot_dessous","Brot-Dessous"],
                                ["brot_plamboz","Brot-Plamboz"],
                                ["corcelles_cormondreche","Corcelles-Cormondrèche"],
                                ["cornaux","Cornaux"],
                                ["cortaillod","Cortaillod"],
                                ["cressier","Cressier"],
                                ["enges","Enges"],
                                ["fresens","Fresens"],
                                ["gorgier","Gorgier"],
                                ["hauterive","Hauterive"],
                                ["brevine","La Brévine"],
                                ["chaux_de_fonds","La Chaux-De-Fonds"],
                                ["chaux_du_milieu","La Chaux-Du-Milieu"],
                                ["cote_aux_fees","La Cote-Aux-Fées"],
                                ["sagne","La Sagne"],
                                ["tene","La Tène"],
                                ["cerneux_pequignot","Le Cerneux-Péquignot"],
                                ["landeron","Le Landeron"],
                                ["locle","Le Locle"],
                                ["brenets","Les Brenets"],
                                ["planchettes","Les Planchettes"],
                                ["ponts_de_martel","Les Ponts-De-Martel"],
                                ["verrieres","Les Verrières"],
                                ["lignieres","Lignières"],
                                ["milvignes","Milvignes"],
                                ["montalchez","Montalchez"],
                                ["neuchatel","Neuchâtel"],
                                ["peseux","Peseux"],
                                ["rochefort","Rochefort"],
                                ["saint_aubin_sauges","Saint-Aubin-Sauges"],
                                ["saint_blaise","Saint-Blaise"],
                                ["val_de_ruz","Val-De-Ruz"],
                                ["val_de_travers","Val-De-Travers"],
                                ["valangin","Valangin"],
                                ["vaumarcus","Vaumarcus"]
                            ]
                        }),
                        valueField: 'myId',
                        displayField: 'displayText'
                    }))
                }
             }
```

**Comment by kalbermattenm** [comment]
+1, that did the trick !

Thanks @sbrunner 

For me, this issue can be closed as soon as #972 and #973 are merged.
